tokio/src/net/unix/stream.rs

```diff
@@ -66,7 +72,20 @@ impl UnixStream {
     where
         P: AsRef<Path>,
     {
-        let stream = mio::net::UnixStream::connect(path)?;
+        // On linux, abstract socket paths need to be considered.
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        let addr = {
+            let os_str_bytes = path.as_ref().as_os_str().as_bytes();
+            if os_str_bytes.starts_with(b"\0") {
+                StdSocketAddr::from_abstract_name(os_str_bytes)?
```

Comment:
Same as https://github.com/tokio-rs/tokio/pull/6772/files#r2335889347 if my analysis is correct